core/src/main/java/org/testcontainers/containers/Network.java

```diff
@@ -0,0 +1,107 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.command.CreateNetworkCmd;
+import lombok.*;
+import org.junit.rules.ExternalResource;
+import org.junit.rules.TestRule;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.utility.ResourceReaper;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+public interface Network extends AutoCloseable, TestRule {
+
+    String getId();
+
+    String getName();
+
+    Boolean getEnableIpv6();
+
+    String getDriver();
+
+    boolean isCreated();
+
+    default boolean create() {
+        return getId() != null;
+    }
+
+    @Override
+    default void close() {
+        if (isCreated()) {
+            ResourceReaper.instance().removeNetworks(getName());
+        }
+    }
+
+    @SneakyThrows
+    @SuppressWarnings("unchecked")
+    default <T extends Network> T as(Class<T> type) {
```

Comment:
I'm curious to that purpose of this class, which is related to another question I had: why have an interface? Do we plan on users extending and supplying their own Network implementation?